src/common/marlin_vars.c

```diff
@@ -62,7 +62,7 @@ marlin_var_id_t marlin_vars_get_id_by_name(const char *var_name) {
     for (int i = 0; i <= MARLIN_VAR_MAX; i++)
         if (strcmp(var_name, __var_name[i]) == 0)
             return i;
-    return -1;
+    return MARLIN_VAR_MAX + 1;
```

Comment:
Isn't it better to define `typedef enum : int {...} marlin_var_id_t;` instead? Or use std::optional?

Reply:
The enum is used as a mask in several other places. In that case I'm a bit against using a signed value.

I agree it's not nice (especially if you look around, we have many + 1 places around this constant), but right now this is just fixing the behavior, no attempt at improving it.